README.md

```diff
@@ -173,6 +173,20 @@ ignores: [
 
 In this way, you can control which files the UI5 linter should process and which it should ignore.
 
+## Internals
+
+UI5 linter makes use of the [TypeScript compiler](https://github.com/microsoft/TypeScript/) to parse and analyze the source code (both JavaScript as well as TypesScript) of a UI5 project. This allows a decent level of accuracy and performance.
+
+For this UI5 linter provides the TypeScript compiler with the SAPUI5 [type definitions](https://sap.github.io/ui5-typescript/), the compiler can then infer the usage of UI5 classes and modules in the code. UI5 linter only needs to ask the compiler whether a given function or property is deprecated and the compiler will provide it with the full deprecation text.
+
+In some cases UI5 linter will transpile some of the source files before analyzing them. For example the `sap.ui.require` syntax is transpiled to `import` statements (basically ESM) so that the TypeScript compiler can understand them.
+
+Similarly, UI5 XML views are transpiled to a JavaScript representation that can be analyzed in the same way as regular JavaScript code. In both cases, source maps are used to map any findings back to the correct line and column position in the original source code.
+
+There are additional checks build on top of the compiler API. For example the use of global variables provided by the UI5 framework (e.g. `sap.m.Button`), or for the correct usage of asynchronous component initialization.
```

Comment:
maybe add something like 
...or checks in general that operator on non-JS/TS code (manifest.json, HTML)

Reply:
Done